### PR TITLE
Leave default SNI on API 24+ 

### PR DESCRIPTION
**Pull request by yschimke**
On Android 24+, SNI is supported, so avoid overriding anything set manually.

### CODE REVIEW

**Comment by yschimke** [comment]
Going to change this to test with the suggested setup

File: android-test/src/androidTest/java/okhttp/android/test/sni/SniOverrideTest.kt [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Request
+import org.junit.jupiter.api.Assumptions.assumeTrue
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Tag
+import org.junit.jupiter.api.Test
+
+/**
+ * Test for new Let's Encrypt Root Certificate.
```
Comment (swankjesse) [inline android-test/src/androidTest/java/okhttp/android/test/sni/SniOverrideTest.kt:36]: I don’t follow this?

File: android-test/src/androidTest/java/okhttp/android/test/sni/SniOverrideTest.kt [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  @Test
+  fun getWithDns() {
+    client = client.newBuilder()
+      .dns {
+        Dns.SYSTEM.lookup("sni.cloudflaressl.com")
+      }
+      .build()
+
+    val request = Request.Builder()
+      .url("https://cloudflare-dns.com/cdn-cgi/trace")
```
Comment (swankjesse) [inline android-test/src/androidTest/java/okhttp/android/test/sni/SniOverrideTest.kt:104]: oooh that’s a neat endpoint

**Review by swankjesse — APPROVED** [review]
MockWebServer should capture include inbound SNI information!

**Comment by swankjesse** [comment]
SNI in MockWebServer:
https://github.com/square/okhttp/pull/7597

File: android-test/src/androidTest/java/okhttp/android/test/sni/SniOverrideTest.kt (hunk 1 above)
Comment (yschimke) [reply android-test/src/androidTest/java/okhttp/android/test/sni/SniOverrideTest.kt:36]: Copy paste bug, I'll fix.

Comment (yschimke) [reply android-test/src/androidTest/java/okhttp/android/test/sni/SniOverrideTest.kt:36]: Copy paste bug, I'll fix.

**Comment by swankjesse** [comment]
LGTM

**Pull request merged**